src/common/io.ts

```diff
@@ -252,11 +281,36 @@ export class IOReadable implements AoIORead {
     });
   }
 
-  async getGateways(): Promise<
-    Record<string, AoGateway> | Record<string, never>
-  > {
-    return this.process.read<Record<string, AoGateway>>({
-      tags: [{ name: 'Action', value: 'Gateways' }],
+  async getGateways(
+    pageParams?: PaginationParams,
+  ): Promise<PaginationResult<AoGateway & { gatewayAddress: WalletAddress }>> {
+    const params = {
+      page: 1,
+      pageSize: 100,
+      sortBy: 'gatewayAddress',
+      sortOrder: 'asc',
+      ...pageParams,
+    };
+
+    const alTags = [
```

Comment:
typo: allTags

Reply:
updated in [00739c0](https://github.com/ar-io/ar-io-sdk/pull/138/commits/00739c010b1b9b0766e302614f2c75933b0a2c4a)